src/main/java/com/sucy/skill/listener/ShieldBlockListener.java

```diff
@@ -0,0 +1,43 @@
+package com.sucy.skill.listener;
+
+import com.sucy.skill.api.event.PlayerBlockDamageEvent;
+import org.bukkit.Bukkit;
+import org.bukkit.Statistic;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.player.PlayerStatisticIncrementEvent;
+
+import java.util.HashMap;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Listener to throw custom {@link PlayerBlockDamageEvent}
+ */
+public class ShieldBlockListener extends SkillAPIListener {
+
+    private final HashMap<Player, EntityDamageByEntityEvent> volatileMap = new HashMap<>();
+
+    /**
+     * Remember EntityDamageByEntityEvent in a moment
+     */
+    @EventHandler
+    public void onDamage(EntityDamageByEntityEvent event){
+        if (!(event.getEntity() instanceof Player)) return;
```

Comment:
Perhaps it may also be good to check if the player [isBlocking](https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/entity/HumanEntity.html#isBlocking()) before adding them to the volatile map and starting a timer?

Reply:
Perhaps then the performance will be better. But what I need is to know how much damage has been blocked

Reply:
That's a good idea